src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexNodes.cs

```diff
@@ -630,10 +630,9 @@ internal sealed class RegexAlternationNode : RegexExpressionNode
         public RegexAlternationNode(RegexAlternatingSequenceList sequenceList)
             : base(RegexKind.Alternation)
         {
+            Debug.Assert(sequenceList.NodesAndTokens.Length > 0);
```

Comment:
A bit more restrictive assert.

```suggestion
            Debug.Assert(sequenceList.NodesAndTokens.Length % 2 == 1);
```